src/com/lftechnology/java/training/alina/collections/generics/wildcards/CustomerUtil.java

```diff
@@ -0,0 +1,75 @@
+
+package com.lftechnology.java.training.alina.collections.generics.wildcards;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * CustomerUtil Class consists of comparisions of different customer
+ * 
+ * @author Alina Shakya <alinashakya@lftechnology.com>
+ * @param <T>
+ */
+public class CustomerUtil<T extends Customer> {
+
+	private T customer;
+
+	public CustomerUtil(T obj) {
+
+		this.setCustomer(obj);
+	}
+
+	public T getCustomer() {
+
+		return customer;
+	}
+
+	public void setCustomer(T customer) {
+
+		this.customer = customer;
+	}
+
+	public float getBill() {
+
+		return customer.getBill();
+	}
+
+	/**
+	 * Generic method to list customers
+	 * 
+	 * @param buttonList
+	 * @return customer {@link List} lists of customers
+	 * @author Alina Shakya <alinashakya@lftechnology.com>
+	 */
+	public static List<CustomerUtil<?>> getCustomers(
+		List<CustomerUtil<?>> customerList) {
+
+		List<CustomerUtil<?>> customers =
+			new ArrayList<CustomerUtil<?>>(customerList.size());
+		for (CustomerUtil<?> customer : customerList) {
+			customers.add(customer);
+		}
+		return customers;
+	}
+
+	/**
+	 * Checks if the salary is equal or not
+	 * 
+	 * @param otherCustomer
+	 * @return
+	 * @author Alina Shakya <alinashakya@lftechnology.com>
+	 */
+	public boolean isBillPaymentEqual(CustomerUtil<?> otherCustomer) {
+
+		if (customer.getBill() == otherCustomer.getBill()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Equality tests should not be made with floating point values. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1244)